### ISSUE TITLE
[Screen reader] Missing aria details for shadow blocks in some cases

### ISSUE THREAD

**Issue by microbit-robert** [RaspberryPiFoundation/blockly#9456]
### Check for duplicates

- [x] I have searched for similar issues before opening a new one.

### Description

There is currently a [check for whether a block is a simple reporter with a full block field](https://github.com/google/blockly/blob/add-screen-reader-support-experimental/core/block_svg.ts#L235-L238), however, this code only checks the first field in the input. There are instances in MakeCode where it is not the first field that is the full block field. Please see the example below where the first field is just the text "melody" and the second field is the full block field.

<img width="405" height="380" alt="Image" src="https://github.com/user-attachments/assets/6846ebe5-db26-45ff-8622-164251864778" />

I think that at least for shadow blocks, the check needs to extend to all of the fields. This same change would also have to be applied to [`getFocusableElement` where the same check is made](https://github.com/google/blockly/blob/add-screen-reader-support-experimental/core/block_svg.ts#L1965-L1971).

### Reproduction steps

1. 
2. 
3.

### Stack trace

```shell

```

**Comment by BenHenning** [comment]
I'm assuming that in this case 'melody' is a field label. We could possibly extend the definition of simple reporter (at least for the sake of screen reader & keyboard navigation, not sure about generally, but maybe that makes sense) to be a block that contains exactly one editable field rather than having exactly one field that's editable. I think that will address the problem here, but some additional considerations need to be made if it's affecting more than keyboard nav.

**Comment by BenHenning** [comment]
This is mostly fixed from #9484 but there's still a change needed in the plugin.

**Comment by BenHenning** [comment]
Fully addressed in https://github.com/RaspberryPiFoundation/blockly-keyboard-experimentation/pull/773.
